### PR TITLE
Adb sync

### LINKED ISSUE #8: Automatic download/upload of the latest database backup using `adb`

### PR DESCRIPTION
**Pull request by larnaa**
Close #8

### CODE REVIEW

File: src/bluecoins_cli/adb.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+from adbutils import adb  # type: ignore
+from adbutils._device import AdbDevice  # type: ignore
+
+
+class Device:
+    def __init__(self, device: AdbDevice) -> None:
+        self.device = device
+
+        self.APP_ID = 'com.rammigsoftware.bluecoins'
```
Comment (droserasprout) [inline src/bluecoins_cli/adb.py:12]: Should be a module-level constant, since it's not modified later.

File: src/bluecoins_cli/adb.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        current_time = datetime.datetime.now().strftime("%Y-%m-%d_%H:%M:%S")
+        self.DB = f"bluecoins-{current_time}"
```
Comment (droserasprout) [inline src/bluecoins_cli/adb.py:15]: Database name should be generated at the moment of its creation.

File: src/bluecoins_cli/adb.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    def pull_db(self) -> None:
+        subprocess.run(
+            f'adb shell su {self.APP_ID} -c "cat /data/user/0/{self.APP_ID}/databases/bluecoins.fydb" > {self.DB}.fydb',
+            shell=True,
+            check=True,
+        )
+
+    def cli_command_run(self, cli_command: str) -> None:
+        subprocess.run(f'poetry run bluecoins-cli {self.DB}.fydb {cli_command}', shell=True, check=True)
```
Comment (droserasprout) [inline src/bluecoins_cli/adb.py:46]: Smart move! But don't rely on Poetry here. You have two options:

* Call `bluecoins-cli` as a script with `subprocess`
* `from bluecoint_cli import cli`, then call it directly

File: src/bluecoins_cli/adb.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+    def push_db_root(self) -> None:
+
+        self.device.sync.push(f'{self.DB}.new.fydb', f'/data/local/tmp/{self.DB}.new.fydb')
+        self.device.shell(
+            f'su 0 -c mv /data/local/tmp/{self.DB}.new.fydb /data/user/0/{self.APP_ID}/databases/bluecoins.fydb'
```
Comment (droserasprout) [inline src/bluecoins_cli/adb.py:52]: Please, add a "FIXME: ..." comment above all lines with privilege escalation.

**Review by droserasprout — CHANGES_REQUESTED** [review]

File: src/bluecoins_cli/adb.py (hunk 2 above)
Comment (larnaa) [reply src/bluecoins_cli/adb.py]: I did this, but not sure in the way of resolve.

Comment (larnaa) [reply src/bluecoins_cli/adb.py]: @droserasprout

**Review by droserasprout — APPROVED** [review]

**Pull request merged**